### ISSUE TITLE
What RIS fields to use so that a [Source] link appears?

### ISSUE THREAD

**Issue by glorious1** [dsifford/academic-bloggers-toolkit#395]
Not an issue, just a question.  I have some references imported via RIS file appearing in the bibliography with a [Source] link, some with a DOI link that shows spelled out, and some that have the DOI link and the [Source] link, even when they are the same.  And some show no link at all.

I'm trying to get them to be consistent.  It seem the UR field goes into [Source] link (which doesn't appear in any form when you double-click on a book reference in ABT).  If I put a random URL in there (http://abc.com) it seems to show up consistently.  However, if I paste a DOI link there (https://doi.org/10.1016/j.foreco.2009.03.024), it usually doesn't show, although I have rarely gotten it to work; I'm not sure how.

FYI, there are some references when I WOULD like to show both a DOI link spelled out and a [Source], for instance when there is a pdf posted on the author's web page.  That seems to work pretty well.

**Comment by dsifford** [comment]
Hm.. Might be a bug somewhere. What you're describing is not the intended effect.

You're correct that `UR` is generally the field you'll want for URLs.

Can you perhaps share the file that is giving you these issues so I can investigate?

Thanks

**Comment by glorious1** [comment]
Sure, I attached a test1.ris (had to change to .txt).  I shaved it down to 6 refs for examples, and have been testing it.  My ideal is, when there is only a DOI source, to put it in the [Source] link, not as a spelled out doi.  If there is also a different link to the pdf somewhere, then I hope to show both.

Bethers has a DO entry and a DOI hyperlink in the UR field.  The DOI gets used to make a spelled-out link.  The UR one doesn't show.  If I delete the DOI entry (manually in ABT), the UR entry still does not appear.

Chappelka is set up the same way with both the DO and UR entry (a link to the DOI).  But here, both show.  So not consistent with the Bethers reference.

Buma has both entries also, but the UR entry is a regular URL, not to a DOI.  Both show, and that's good.

Feucht just has a non-DOI UR and it works fine.

Skelly is a book.  The RIS has a UR entry for it, and it shows up in the References as a [Source].  All good, but when you manually edit a book in ABT, there is no URL field.  Not very important.

Now try an experiment.  Manually in ABT, open Chappelka, copy the URL (https:doi.org . . .) and past it into another reference, like Buma, replacing the existing URL.  It doesn't show.  It works in Chappelka, but not when you paste it into Buma's URL field.  In fact, I have never gotten any doi-type URL to show after manually putting it in in ABT.  Even the ones that come from RIS don't always work, as you see in comparing Bethers and Chappelka.

Sorry to drag you through all this!

[test1.txt](https://github.com/dsifford/academic-bloggers-toolkit/files/1565168/test1.txt)
